src/Language/Fixpoint/Solver/PLE.hs

```diff
@@ -653,6 +653,26 @@ eval γ ctx et = go
 -- | 'evalELamb' produces equations that preserve the context of a rewrite
 -- so equations include any necessary lambda bindings.
 evalELam :: Knowledge -> ICtx -> EvalType -> (Symbol, Sort) -> Expr -> EvalST (Expr, FinalExpand)
+evalELam γ ctx et (x, s) e
+  | not $ isEtaSymbol x = do
+    -- We need to refresh it as for some reason lamdba variables
+    -- as reflected lambdas variables for some reason are declared
+    -- two times, maybe we should define a new type of identifier
+    -- and not reuse the etabeta ones
```

Comment:
Can we have an example of what it means for a lambda variable to be reflected twice?

It looks like it would be worth investigating that, instead of changing the code here to workaround a possible bug.

Reply:
Idk why but if for example you take the SKILam and remove this branch it will complain that the variable name that is used by the lambdas is declared twice in the smt file

Reply:
Tried to reproduce this in a smaller example, but couldn't. But I think it is ok to investigate it separately.